src/components/buttons/icon-button/icon-button.js

```diff
@@ -33,15 +33,17 @@ const getIconThemeColor = props => {
 };
 
 export const IconButton = props => {
-  const buttonAttributes = {
+  const attributes = {
     'data-track-component': 'IconButton',
-    ...filterAriaAttributes(props),
-    ...filterDataAttributes(props),
+    ...filterInvalidAttributes(props),
   };
+
   const isActive = props.isToggleButton && props.isToggled;
+
   return (
     <AccessibleButton
-      buttonAttributes={buttonAttributes}
+      as={props.as}
+      buttonAttributes={attributes}
```

Comment:
I have a concern about forwarding all props, given we already "reimplement" some of the HTML props:
- It is possible to pass both the props `disabled` and `isDisabled` to a component, and in this case, because the [`AccessibleButton`](https://github.com/commercetools/ui-kit/blob/67dcdef9397d6a37ba1e2d268489ab1df21397fa/src/components/buttons/accessible-button/accessible-button.js) spreads the `buttonAttributes` prop _after_ the `isDisabled` prop is assigned to `disabled`, which means the native html prop will override the `isDisabled` one. Same for all other props that we manually assign before the `buttonAttributes` spread. Is this an intended behaviour?

Reply:
I don't get it. You mean if the user passed in `disabled={true}` and `isDisabled={false}`?

Reply:
Kind of feels like a play stupid games win stupid prizes situation.

Reply:
Oh nice catch!

Reply:
What would you prefer to happen? I guess we could add a prop-type warning if a user passes both?

Reply:
Yeah, it shouldn't happen, but it _can_ lead to weird bugs that are hard to track.
It's more a question of: should we worry about it, and can we do something to mitigate this?

Reply:
I would just ignore it. The plan is to deprecate all the `isDisabled`, etc stuff right? 

If someone writes the code
```
<IconButton disabled={true} isDisabled={false} />
```

and it passes pull request, then we deserve the bugs.

Reply:
Yeah, it totally makes sense to deprecate the `isDisabled` and others as the next step.

It could happen accidentally after some layers of abstraction, but I also wouldn't worry about it too much, as for now we're still supposed to use the explicit props that are documented.

I'm OK with keeping it as it is, as long as we're aware and we hopefully deprecate those redundant props soon 😇.

Reply:
Having two "sources of truth" is a bad idea. 😄

Reply:
One could have like a prop-zipper or something so that always one has precedence. However, I think it's a super nice theoretical catch but given the scale we are the practical implications can be negated.